tkt-70977: Add interactive upgrade variable

skarekrow opened the pull request:
This will sidestep a lot of issues that came up from the one stdin mode we had for noninteractive cases.

- Add InteractiveExec class
- Workaround for master issues with fetching freebsd-update
- No longer fetch master freebsd-update scripts for upgrade, but instead grab the target RELEASEs version of the script
- Add interactive switch and change behavior depending on mode
- Remove a lot of unneeded code now from interactive commands (from iocage API side) and console.

FreeNAS Ticket: #70977
Closes #771
Closes #769

Commit:
Add interactive upgrade variable

This will sidestep a lot of issues that came up from the one stdin mode we had for noninteractive cases.

- Add InteractiveExec class
- Workaround for master issues with fetching freebsd-update
- No longer fetch master freebsd-update scripts for upgrade, but instead grab the target RELEASEs version of the script
- Add interactive switch and change behavior depending on mode
- Remove a lot of unneeded code now from interactive commands (from iocage API side) and console.

FreeNAS Ticket: #70977
Closes #771
Closes #769

gronke commented on iocage_lib/ioc_upgrade.py:
```diff
@@ -120,21 +125,62 @@ def upgrade_jail(self):
+                    pass
+            else:
+                # FreeBSD update loops 3 times
+                for _ in range(3):
```
What is the reason to run exactly 3 attempts to upgrade a jail?

igalic replied:
this document contains exactly three `freebsd-update install` in its procedure https://www.freebsd.org/doc/en/books/handbook/updating-upgrading-freebsdupdate.html

dgeo commented:
Le 19 janvier 2019 09:40:15 GMT+01:00, "Stefan Grönke" <notifications@github.com> a écrit :
>gronke commented on this pull request.
>
>
>
>> +                    iocage_lib.ioc_common.logit(
>+                        {
>+                            "level": "EXCEPTION",
>+                            "message": msg
>+                        },
>+                        _callback=self.callback,
>+                        silent=self.silent
>+                    )
>+
>+
>+            if not self.interactive:
>+                while not self.__upgrade_install__(tmp.name):
>+                    pass
>+            else:
>+                # FreeBSD update loops 3 times
>+                for _ in range(3):
>
>What is the reason to run exactly 3 attempts to upgrade a jail?

First ``freebsd-update install`` will install kernel, second run installs 'world', the third removes old files and libs (only for major upgrades in my experience)

-- 
geoffroy desvernay (mobile)

Commit:
Fix handful of issues with exec

- {jail|host}_user not being respected
- Skip the starting messages when using interactiveexec in API
- Fix tests to reflect that jails will be started for exec

Commit:
Travis

Commit:
Fix clone jail tests
